src/cargo/util/config/value.rs

```diff
@@ -78,12 +78,7 @@ impl Definition {
     ///
     /// CLI is preferred over environment, which is preferred over files.
     pub fn is_higher_priority(&self, other: &Definition) -> bool {
-        match (self, other) {
-            (Definition::Cli, Definition::Environment(_)) => true,
-            (Definition::Cli, Definition::Path(_)) => true,
-            (Definition::Environment(_), Definition::Path(_)) => true,
-            _ => false,
-        }
+        matches!((self, other), (Definition::Cli, Definition::Environment(_)) | (Definition::Cli, Definition::Path(_)) | (Definition::Environment(_), Definition::Path(_)))
```

Comment:
This [came up previously](https://github.com/rust-lang/cargo/pull/8831#discussion_r517432507), but I don't think this change should be made.